redaxo/src/core/lib/form/form_base.php

```diff
@@ -1059,7 +1064,8 @@ protected function validate()
                 }
 
                 $validator = $element->getValidator();
-                if (!$validator->isValid($element->getSaveValue())) {
+                $saveValue = $element->getSaveValue();
+                if ($saveValue !== null && !$validator->isValid((string) $saveValue)) {
```

Comment:
@gharlan passt das?

Reply:
Ich weiß nicht genau, wann null vorkommt, aber ich denke, es passt eher nicht.
Denn wenn man eine `notEmpty`-Validierung hinzugefügt hat, und $saveValue null ist, dann bekommt man jetzt keinen Fehler mehr, die Validierung greift also nicht.

Entweder castest du hier alles nach string, auch `null`, oder wir passen den validator so an, dass er nicht nur strings entgegen nimmt (bei den einzelnmethoden in rex_validator habe ich das teils schon gemacht).

Reply:
Dann sind die phpdocs im validator aber alle falsch wenn dort auch null als value erlaubt ist 

https://github.com/redaxo/redaxo/blob/dfae9ba5d471e7d5b153b708bca162778194e33a/redaxo/src/core/lib/util/validator.php#L96

Reply:
> Dann sind die phpdocs im validator aber alle falsch wenn dort auch null als value erlaubt ist .. sollte der validator dann auch bool|int etc unterstützen?

Insgesamt bin ich mir unsicher, ob wir den validator ändern sollten, oder hier die Stelle nur.
Und wenn wir den Validator ändern, wie wir ihn dann genau ändern.

Aber wie wäre es, wenn wir hier alles nach string casten, auch null?

> > bei den einzelnmethoden in rex_validator habe ich das teils schon gemacht
> 
> Wo?

Oh, ich habe mich vertan und es verwechselt mit `rex_formatter`. Da hatte ich an manchen Stellen neu auch ints zugelassen, nicht nur strings.